### ISSUE TITLE
Imports used in "# type" are detected as unused in 6.0.0

### ISSUE THREAD

**Issue by SamuilDichev** [PyCQA/flake8#1755]
### how did you install flake8?

```console
$ pip install flake8==6.0.0
```

### unmodified output of `flake8 --bug-report`

```json
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.8.10",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.10.0"
    },
    {
      "plugin": "pyflakes",
      "version": "3.0.0"
    }
  ],
  "version": "6.0.0"
}
```

#### what I expected to happen

Imports used only for type hinting via `# type` to be detected as **used** and not be highlighted by flake8 as unused.

#### sample code

```python
from typing import Dict
gaz = {}  # type: Dict[int, int]
```

#### commands ran

```console
$ flake8 test/test.py
test/test.py:1:1: F401 'typing.Dict' imported but unused
```

**Comment by sigmavirus24** [comment]
In the future, please search **closed and** open issues before creating new ones that are duplicates.
